docker/dendrite_sytest.sh

```diff
@@ -0,0 +1,63 @@
+#! /usr/bin/env bash
+
+set -ex
+
+# Attempt to find a sytest to use.
+# If /test/run-tests.pl exists, it means that a SyTest checkout has been mounted into the Docker image.
+if [ -e "./run-tests.pl" ]
+then
+    # If the user has mounted in a SyTest checkout, use that. We can tell this by files being in the directory.
+    echo "Using local sytests..."
+else
+    # Otherwise, try and find out what the branch that the Dendrite checkout is using. Fall back to develop if it's not a branch.
+    branch_name="$(git --git-dir=/src/.git symbolic-ref HEAD 2>/dev/null)" || branch_name="develop"
+
+    # Try and fetch the branch
+    echo "Trying to get same-named sytest branch..."
+    wget -q https://github.com/matrix-org/sytest/archive/$branch_name.tar.gz -O sytest.tar.gz || {
+        # Probably a 404, fall back to develop
+        echo "Using develop instead..."
+        wget -q https://github.com/matrix-org/sytest/archive/develop.tar.gz -O sytest.tar.gz
+    }
+
+    tar --strip-components=1 -xf sytest.tar.gz
+
+fi
+
+# Make sure all Perl deps are installed -- this is done in the docker build so will only install packages added since the last Docker build
+dos2unix ./install-deps.pl
+./install-deps.pl
```

Comment:
Can you please also run this in the dockerfile, that way most dependencies will be pre-installed but new ones will still be picked up. Otherwise it takes like 15mins